tests/mocha/block_test.js

```diff
@@ -1680,4 +1680,116 @@ suite('Blocks', function() {
       });
     });
   });
+  suite('toString', function() {
+    var toStringTests = [
```

Comment:
@moniika had some thoughts as I was adding these tests.
The problem at hand was that I needed tests for toString, I could go ahead and recreate block definitions of the blocks I want to tests, similar to what is done [here](https://github.com/google/blockly/blob/master/tests/mocha/astnode_test.js#L9) but that had two drawbacks: 
- In terms of coverage, that screams to me as re-inventing the wheel (surely there's a bunch of tests that we want to cover a whole group of baseline blocks), and we don't want to re-generate what that baseline is every time. 
- Built-in blocks would be much easier to use as I can use the playground to generate the xml needed.

That got me here, where I used a similar pattern adopted in blockly-samples, to create the tests in some intermediate format, and generate them in a forEach. I like this method in that it's less verbose, but this still doesn't address the fact that it would be nice if I didn't have to generate what the baseline set of blocks and their respective xml is every time a test of this nature needs to be added.

As an aside, it was also clunky to have to add the XML without multi-line comments, that's just wasted time there, where we can just enable multi-line comments in mocha tests (after all, we're not running our unit tests in IE).

I think we need some sort of baseline xml definitions, similar to those of screen shot [tests](https://github.com/google/blockly/tree/master/tests/screenshot/test_cases), that we can run actions on top of. In this case, I would add the expected output for each of these tests, and 
update the test runner to run my test for each of these baseline blocks.

@rachel-fenichel FYI

Reply:
+1 to multi-line comments; no comment yet on the rest.

Reply:
For test block definitions, all of the common ones (empty block, value block, statement block) should definitely move into test_helpers.js.  We can either define and undefine them each time (calling helpers) or define them globally for all tests in a wrapper setup function.

I slightly prefer defining and undefining in each suite that needs them, with something like `helpers_define_blocks('empty_block', 'value_block', ...)` and `helpers_undefine_blocks('empty_block', 'value_block', ...)'.

Reply:
Definitely if it seems useful for multiple tests then adding block_name and xml (or function that generates xml allowing for customization of values) could be added to helpers.

We currently have some helpers for defining blocks. I would be in favor of a pattern where you could use helper to define blocks and that would keep track of the blocks that were defined during the test (and then you call a cleanup method that undefines the blocks without needing to remember what you defined). They aren't being used consistently, so cleanup is needed there (and also a convenient way to undefine isn't implemented yet).